pkg/controllers/secretstore/client_manager.go

```diff
@@ -257,3 +265,16 @@ func assertStoreIsUsable(store esv1beta1.GenericStore) error {
 	}
 	return nil
 }
+
+func matchStringInList(list []string, item string) (bool, error) {
+	for _, ll := range list {
+		if item == ll {
```

Comment:
Why do we need to check this separately?

Reply:
I guess the intention was that if the namespace perfectly equals to the regex which could just be the namespace, than there is no need to Compile the regex. Which is weird, because at that point, just use a plain namespace match 🤷 

I was trying to make this a bit more efficient because compiling regexes is actually super slow, but since I can't ties the regexes to the condition ( because it's a list and the list can even be reordered so I can't use the index ) so even a dumb cache does not work or ends up being super ugly code which I didn't like.

So for now, we'll just have to go with it.